components/net/http_loader.rs

```diff
@@ -846,11 +856,20 @@ fn http_network_or_cache_fetch(request: Rc<Request>,
             unreachable!()
     };
 
-    // Step 9
-    if cors_flag ||
-      (*http_request.method.borrow() != Method::Get && *http_request.method.borrow() != Method::Head) {
-        // TODO update this when https://github.com/hyperium/hyper/pull/691 is finished
-        // http_request.headers.borrow_mut().set_raw("origin", origin);
+    // Step 7
+    if http_request.omit_origin_header.get() == false {
+        let method = http_request.method.borrow();
+        if cors_flag || (*method != Method::Get && *method != Method::Head) {
```

Comment:
According to the [spec](https://fetch.spec.whatwg.org/#concept-request-origin), the request's origin will be changed to `"client"` during fetching. We should have a `debug_assert!(*http_request.origin.borrow() != Origin::Client)` and go with `if let` here.